tonic-build/src/manual.rs

```diff
@@ -0,0 +1,444 @@
+    /// Panics if `name`, `route_name`, `input_type`, `output_type`, or `codec_path` weren't set.
+    pub fn build(self) -> Method {
+        Method {
+            name: self.name.unwrap(),
+            route_name: self.route_name.unwrap(),
+            comments: self.comments,
+            input_type: self.input_type.unwrap(),
+            output_type: self.output_type.unwrap(),
+            client_streaming: self.client_streaming,
+            server_streaming: self.server_streaming,
+            codec_path: self.codec_path.unwrap(),
+        }
+    }
+}
+
+struct ServiceGenerator {
+    builder: Builder,
+    clients: TokenStream,
+    servers: TokenStream,
+}
+
+impl ServiceGenerator {
+    fn generate(&mut self, service: &Service) {
+        if self.builder.build_server {
+            let server = server::generate(
+                service,
+                true,  // emit_package,
+                "",    // proto_path, -- not used
+                false, // compile_well_known_types -- not used
+                &Attributes::default(),
+            );
+            self.servers.extend(server);
+        }
+
+        if self.builder.build_client {
+            let client = client::generate(
+                service,
+                true,  // emit_package,
+                "",    // proto_path, -- not used
+                false, // compile_well_known_types, -- not used
+                &Attributes::default(),
+            );
+            self.clients.extend(client);
+        }
+    }
+
+    fn finalize(&mut self, buf: &mut String) {
+        if self.builder.build_client && !self.clients.is_empty() {
+            let clients = &self.clients;
+
+            let client_service = quote::quote! {
+                #clients
+            };
+
+            let ast: syn::File = syn::parse2(client_service).expect("not a valid tokenstream");
+            let code = prettyplease::unparse(&ast);
+            buf.push_str(&code);
+
+            self.clients = TokenStream::default();
+        }
+
+        if self.builder.build_server && !self.servers.is_empty() {
+            let servers = &self.servers;
+
+            let server_service = quote::quote! {
+                #servers
+            };
+
+            let ast: syn::File = syn::parse2(server_service).expect("not a valid tokenstream");
+            let code = prettyplease::unparse(&ast);
+            buf.push_str(&code);
+
+            self.servers = TokenStream::default();
+        }
+    }
+}
+
+/// Service generator builder.
+#[derive(Debug)]
+pub struct Builder {
+    build_server: bool,
+    build_client: bool,
+
+    out_dir: Option<PathBuf>,
+}
+
+impl Default for Builder {
+    fn default() -> Self {
+        Self {
+            build_server: true,
+            build_client: true,
+            out_dir: None,
+        }
+    }
+}
+
+impl Builder {
+    /// Create a new Builder
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Enable or disable gRPC client code generation.
```

Comment:
prob wanna mention the default here? (maybe we missed this also in the prost side).